bound/kt/src/main/kotlin/web5/sdk/crypto/keys/InMemoryKeyManager.kt

```diff
@@ -0,0 +1,43 @@
+package web5.sdk.crypto.keys
+
+import web5.sdk.crypto.signers.Ed25519Signer
+import web5.sdk.rust.InMemoryKeyManager as RustCoreInMemoryKeyManager
+
+/**
+ * A class for managing cryptographic keys in-memory.
+ */
+class InMemoryKeyManager : KeyManager {
+    private val rustCoreKeyManager = RustCoreInMemoryKeyManager()
+
+    /**
+     * Generates new key material and returns the public key represented as a Jwk.
+     *
+     * @return Jwk the public key represented as a Jwk.
+     */
+    override fun generateKeyMaterial(): Jwk {
+        val jwkData = rustCoreKeyManager.generateKeyMaterial()
+        return Jwk.fromBinding(jwkData)
+    }
+
+    /**
+     * Returns the Ed25519Signer for the given public key.
+     *
+     * @param publicKey the public key represented as a Jwk.
+     * @return Ed25519Signer the signer for the given public key.
+     */
+    override fun getSigner(publicKey: Jwk): Ed25519Signer {
+        return Ed25519Signer(publicKey)
```

Comment:
Ah, the signer needs the private key so instead you'll need to make a call to `rustCoreKeyManager.getSigner()` with the public key (which you'll have to first convert to rust core JWK), and then you'll have to implement a `Ed25519Signer.fromBinding(rustCoreSigner)` (or like my prior comment recommended `Ed25519.fromRustCore(rustCoreSigner)`)

So yeah, the `Ed25519Verifier` takes a public key in its constructor and the signer takes a private key

Reply:
So yeah, something like this

```suggestion
        val rustCorePublicKey = publicKey.toRustCore()
        val rustCoreSigner = rustCoreKeyManager.getSigner(rustCorePublicKey)
        return Ed25519Signer.fromRustCore(rustCoreSigner)
```